### PR TITLE
Remove legacy Backend API

### PR DESCRIPTION
**Pull request by rkimballn1**
Removed the runtime::ExternalFunction and runtime::CallFrame and change all Backend API calls to use the new API. The legacy API has been removed.
The transformers are largely unchanged, but classes such as CallFrame no long derive from runtime::CallFrame.

### CODE REVIEW

File: test/ref_generators/generate_convolution_ref.py [hunk 1]
```diff
@@ -210,8 +210,18 @@ def emit_test(t,f):
...
+//
+// DO NOT EDIT THIS FILE. If you want to add new tests, you should edit
+//  test/ref_generators/generate_convolution_ref.py and regenerate this file.
+//
+// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! DO NOT EDIT THIS FILE !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+// !!!!!!!!!!!!!! THIS FILE IS AUTOGENERATED OUTSIDE OF THE BUILD PROCESS !!!!!!!!!!!!!!
 TEST (${BACKEND_NAME}, %s)
 {
+    SKIP_TEST_FOR("GPU", "${BACKEND_NAME}");
+    SKIP_TEST_FOR("NNP_TESTER", "${BACKEND_NAME}");
```
Comment (yxlao) [inline test/ref_generators/generate_convolution_ref.py]: We could enable `NNP_TESTER` here

File: test/util/benchmark.cpp [hunk 2]
```diff
@@ -200,41 +198,37 @@ void run_benchmark(shared_ptr<Function> f,
 {
     stopwatch timer;
     timer.start();
-    auto manager = runtime::Manager::get(backend_name);
-    auto external = manager->compile(f);
-    external->set_emit_timing(timing_detail);
-    auto backend = manager->allocate_backend();
-    auto cf = backend->make_call_frame(external);
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
```
[the remark this answers is not in the document]
Comment (rkimballn1) [reply test/util/benchmark.cpp]: fixed

**Comment by rkimballn1** [comment]
TESTNOW

**Comment by Krovatkin** [comment]
TESTNOW

**Pull request merged**